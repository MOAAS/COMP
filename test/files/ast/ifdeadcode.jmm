class IfDeadCode {
	public boolean veryComplexCalculation() {
	    return true;
	}

	public static void main(String[] args) {
        if(1 < 2 && !(5 * 2 + 1 < 10)) // true
            io.println(1);
        else { // else ignorado
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
        }

        // Tudo a partir daqui é ignorado

        // ultima instrução é io.println(1)
        while(false) { // tudo ignorado
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
        }

        if (new IfDeadCode().veryComplexCalculation()) { }
        else { }

        if (true && 1 < 2) { }
        else {
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
        }

        if (true && !(1 < 2)) {
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
            io.println(999); io.println(999); io.println(999); io.println(999); io.println(999); io.println(999);
        }
        else { }
	}
}