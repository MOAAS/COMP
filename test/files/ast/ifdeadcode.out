Program
 ClassDeclaration
  ClassInfo
   IfDeadCode
  RegularMethodDeclaration
   MethodType
    boolean
   MethodName
    veryComplexCalculation
   MethodVars
   Return
    True
  MainDeclaration
   ArgName
    args
   MethodContent
    Dot
     io
     DotArgs
      println
      1
